pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -64,18 +84,158 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 	reqLogger.Info("Reconciling UserAccount")
 
 	// Fetch the UserAccount instance
-	instance := &toolchainv1alpha1.UserAccount{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	userAcc := &toolchainv1alpha1.UserAccount{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: config.GetOperatorNamespace(), Name: request.Name}, userAcc)
 	if err != nil {
 		if errors.IsNotFound(err) {
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
 			return reconcile.Result{}, nil
 		}
-		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	var created bool
+	var user *userv1.User
+	if user, created, err = r.ensureUser(userAcc); err != nil {
+		_ = r.updateStatus(userAcc, toolchainv1alpha1.StatusProvisioning, err.Error())
```

Comment:
if `r.ensureUser` returns an error, does it really mean that the status should be set to `provisioning`?  Shouldn't that be either `failed` or `unknown`?

Reply:
my understanding is, first step in provisioning is create_user. so we started provisioning and tried to create_user which failed it means that there is error in provisioning due to create_user.

Reply:
it can also fail on getting the User resource (even if it exists)... My point was that when there is an error, then you cannot say for sure that the user is/not provisioned. The status should reflect that... On the other hand, the next reconcile should start pretty quickly, so the next round could fix that.

Reply:
@MatousJobanek I am not getting you point.

What I can say is, when you start `reconcile()` and if userAcc is found then one can/should set status="provisioning".  Later if some error happened then the userAcc.Status.Error should be updated with it and above code does that.  I am not elaborating the reasoning much as we already have another PR for follow-up changes.